core/src/verification/verification.rs

```diff
@@ -44,9 +45,15 @@ pub fn verify_block_basic(header: &Header, bytes: &[u8]) -> Result<(), Error> {
 
     let body_rlp = Rlp::new(bytes).at(1)?;
 
-    for t in body_rlp.iter().map(|rlp| rlp.as_val::<UnverifiedTransaction>()) {
-        t?.verify_basic()?;
-    }
+    let raw_transactions = body_rlp
+        .iter()
+        .map(|rlp| {
+            let tx = rlp.as_val::<UnverifiedTransaction>()?;
+            tx.verify_basic()?;
+            Ok(rlp.as_raw().to_vec())
```

Comment:
I used the transaction hash in here; it should be the raw bytes of transactions.